### PR TITLE
[mellanox|ffb] use system level warm reboot for Mellanox fastfast boot

### PR DESCRIPTION
**Pull request by stepanblyschak**
use common warm boot flow for mellanox fastfast boot.

Signed-off-by: Stepan Blyschak <stepanb@mellanox.com>

### CODE REVIEW

File: syncd/syncd.cpp [hunk 1]
```diff
@@ -3683,56 +3679,77 @@ int syncd_main(int argc, char **argv)
                 FlexCounter::removeAllCounters();
                 stopNotificationsProcessingThread();
 
-                sai_attribute_t attr;
+                /* FIXME: Temprary W/A */
```
Comment (qiluo-msft) [inline syncd/syncd.cpp]: >Temprary [](start = 26, length = 8)

Typo

File: syncd/syncd.cpp [hunk 2]
```diff
@@ -3683,56 +3679,77 @@ int syncd_main(int argc, char **argv)
...
+                char* platform = getenv("platform");
+                if (platform && (strncmp(platform, "mellanox", 8) == 0))
```
Comment (qiluo-msft) [inline syncd/syncd.cpp]: >8 [](start = 63, length = 1)

refactor to remove the magic number

File: syncd/syncd.cpp [hunk 3]
```diff
@@ -3771,6 +3788,15 @@ int syncd_main(int argc, char **argv)
         {
             SWSS_LOG_NOTICE("Warm Reboot requested, keeping data plane running");
 
+
+            /* FIXME: W/A */
+            char* platform = getenv("platform");
+            if (platform && (strncmp(platform, "mellanox", 8) == 0))
```
Comment (qiluo-msft) [inline syncd/syncd.cpp]: >8 [](start = 59, length = 1)

magic number

File: syncd/syncd.cpp (hunk 1 above)
Comment (qiluo-msft) [inline syncd/syncd.cpp]: >W/A [](start = 35, length = 3)

Could you elaborate why this is a W/A, and what is your future plan?

File: syncd/syncd.cpp [hunk 4]
```diff
@@ -3771,6 +3788,15 @@ int syncd_main(int argc, char **argv)
         {
             SWSS_LOG_NOTICE("Warm Reboot requested, keeping data plane running");
 
+
+            /* FIXME: W/A */
```
Comment (qiluo-msft) [inline syncd/syncd.cpp]: >FIXME: W/A [](start = 15, length = 10)

future plan?

File: syncd/Makefile.am [hunk 5]
```diff
@@ -37,6 +37,10 @@ syncd_CPPFLAGS += -DSAITHRIFT=yes
 syncd_LDADD += -lrpcserver -lthrift
 endif
 
+if sonic_asic_platform_mellanox
+syncd_CPPFLAGS += -DUNINIT_DATA_PLANE_ON_REMOVAL
```
Comment (qiluo-msft) [inline syncd/Makefile.am]: >UNINIT_DATA_PLANE_ON_REMOVAL [](start = 20, length = 28)

suggest macro name: SAI_SUPPORT_UNINIT_DATA_PLANE_ON_REMOVAL
to indicate this is a feature, not a specific action.

The UNINIT_DATA_PLANE_ON_REMOVAL may indicate that you want to destroy data plane on removal for Mellanox ASIC.

**Comment by liat-grozovik** [comment]
As all comments were related to code that is currently removed i see no reason why not to approve and merge.
@qiluo-msft  and @kcudnik  can you please review again and approve?

**Pull request merged**